apps/server/src/modules/task/uc/task.uc.ts

```diff
@@ -1,129 +1,91 @@
 
 		return [computedTasks, total];
 	}
 
-	async findAllCompletedForStudent(
-		userId: EntityId,
-		pagination: IPagination
-	): Promise<Counted<TaskWithSubmissionStatus[]>> {
-		const parents = await this.findPermittedTaskParents(userId, Permission.read);
-
-		const [submissionsOfStudent] = await this.submissionRepo.findAllByUserId(userId);
-		const taskIdsWithSubmissions = [...new Set(submissionsOfStudent.map((submission) => submission.task.id))];
+	private async findAllForTeacher(userId: EntityId, pagination: IPagination): Promise<Counted<TaskWithStatusVo[]>> {
+		const parentIds = await this.authorizationService.getPermittedCourses(userId, TaskParentPermission.write);
 
-		const parentIds = parents.map((course) => course.id);
-
-		const [tasks, total] = await this.taskRepo.findAllCurrentByIds(parentIds, taskIdsWithSubmissions, {
+		const [tasks, total] = await this.taskRepo.findAll(parentIds, {
 			pagination,
-			order: { dueDate: SortOrder.asc },
+			order: { createdAt: SortOrder.desc },
 		});
 
-		const domain = new TaskDomainService(tasks, parents);
-		const computedTasks = domain.computeStatusForStudents(submissionsOfStudent);
+		const computedTasks = tasks.map((task) => this.computeTaskStatusForTeacher(task));
 
 		return [computedTasks, total];
 	}
 
-	// TODO: rename teacher and student
-	async findAllOpenForTeacher(userId: EntityId, pagination: IPagination): Promise<Counted<TaskWithSubmissionStatus[]>> {
-		const parents = await this.findPermittedTaskParents(userId, Permission.write);
-
-		const parentIds = parents.map((parent) => parent.id);
+	private hasTaskDashboardPermission(currentUser: ICurrentUser, permission: TaskDashBoardPermission): boolean {
+		const hasPermission = currentUser.user.permissions.includes(permission);
+		return hasPermission;
+	}
 
-		const [tasks, total] = await this.taskRepo.findAll(parentIds, { pagination, order: { createdAt: SortOrder.desc } });
-		const [submissionsOfTeacher] = await this.submissionRepo.findAllByTaskIds(tasks.map((o) => o.id));
+	private computeTaskStatusForStudent(task: Task, userId: EntityId): TaskWithStatusVo {
+		const studentSubmissions = task.submissions.getItems().filter((submission) => submission.student.id === userId);
 
-		const domain = new TaskDomainService(tasks, parents);
-		const computedTasks = domain.computeStatusForTeachers(submissionsOfTeacher);
+		const submitted = studentSubmissions.length > 0 ? 1 : 0;
+		const graded = studentSubmissions.filter((submission) => submission.isGraded()).length;
+		const maxSubmissions = 1;
 
-		return [computedTasks, total];
+		return new TaskWithStatusVo(task, { submitted, maxSubmissions, graded });
 	}
 
-	async findAllOpen(currentUser: ICurrentUser, pagination: IPagination): Promise<Counted<TaskWithSubmissionStatus[]>> {
-		let response: Counted<TaskWithSubmissionStatus[]>;
+	private computeTaskStatusForTeacher(task: Task): TaskWithStatusVo {
```

Comment:
What is mean with Vo?

Reply:
that is a value object. it carries the status information. that is basically just a class representing the former `ISubmissionStatus` interface:
https://github.com/hpi-schul-cloud/schulcloud-server/blob/85ab347be31ab21c4b505f12c75e0c55e4d82dab/apps/server/src/modules/task/domain/TaskDomainService.ts#L6

Reply:
ok